hbase-server/src/main/java/org/apache/hadoop/hbase/compactionserver/CompactionThreadManager.java

```diff
@@ -150,6 +150,27 @@ public void requestCompaction(CompactionTask compactionTask) throws IOException
     }
   }
 
+  /**
+   * Open store, and clean stale compacted file in cache
+   */
+  private HStore openStore(RegionInfo regionInfo, ColumnFamilyDescriptor cfd, boolean major,
```

Comment:
Is this related to this issue or just a refactoring?

Reply:
not just refactor, we need setFavoredNodes on store.
So move this logic outof `selectCompaction` method